### PR TITLE
Selflink is added to all public resource properties.

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

Adding the self link to all the public resource properties. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs [hunk 1]
```diff
@@ -147,6 +147,7 @@ public async Task PartitionedCRUDTest()
             string diagnostics = containerResponse.Diagnostics.ToString();
             Assert.IsFalse(string.IsNullOrEmpty(diagnostics));
             Assert.IsTrue(diagnostics.Contains("StatusCode"));
+            Assert.IsNotNull(containerResponse.Resource.SelfLink);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Can we please assert on #parts across?

**Comment by kirankumarkolli** [comment]
Please include the issue it closes.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/AccountProperties.cs [hunk 2]
```diff
@@ -78,6 +78,17 @@ internal AccountProperties()
...
+        /// <summary>
+        /// Gets the self-link associated with the resource from the Azure Cosmos DB service.
+        /// </summary>
+        /// <value>The self-link associated with the resource.</value> 
+        /// <remarks>
+        /// A self-link is a static addressable Uri for each resource within a database account and follows the Azure Cosmos DB resource model.
+        /// E.g. a self-link for a document could be dbs/db_resourceid/colls/coll_resourceid/documents/doc_resourceid
+        /// </remarks>
+        [JsonProperty(PropertyName = Constants.Properties.SelfLink)]
+        public string SelfLink { get; private set; }
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/AccountProperties.cs]: @kirankumarkolli should this be removed? For account properties it seems to always be empty string.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/AccountProperties.cs]: Yup its unexpected :-( please exclude it. 
Nice catch.

**Pull request merged**